metadata-ingestion/src/datahub/ingestion/sink/datahub_rest.py

```diff
@@ -34,6 +36,10 @@
 
 logger = logging.getLogger(__name__)
 
+DEFAULT_REST_SINK_MAX_THREADS = int(
+    os.getenv("DATAHUB_REST_SINK_DEFAULT_MAX_THREADS", 15)
```

Comment:
Shouldn't this be set based on the number of cores?

Reply:
Because of the python GIL, python pretty much runs on a single core regardless - the total number of cores available only matters when using multiprocessing, not threading.

The threads help because the rest sink is usually bound by I/O, which releases the GIL and allows other threads to run